Add info on using --use-ssh-agent

tahliar opened the pull request:
### PR creator: Description

Added a section on logging in, and other bits and pieces.


### PR creator: Are there any relevant issues/feature requests?

* jsc#PED-7399


### PR creator: Which product versions do the changes apply to?

When opening a PR, check all versions of the documentation that your PR applies to.

- SLE-HA 15
  - [x] 15 next *(current `main`, no backport necessary)*
  - [ ] 15 SP5
  - [ ] 15 SP4
  - [ ] 15 SP3
  - [ ] 15 SP2
  - [ ] 15 SP1
- SLE-HA 12
  - [ ] 12 SP5
  - [ ] 12 SP4

### PR reviewer only: Have all backports been applied?

The doc team member merging your PR will take care of backporting to older documents.
When opening a PR, do *not* set the following check box.

- [ ] all necessary backports are done

tahliar commented:
Admin guide: I removed the user privileges note (https://documentation.suse.com/sle-ha/15-SP5/html/SLE-HA-all/cha-ha-config-basics.html#cha-ha-manual-config) and replaced it with this: 

![image](https://github.com/SUSE/doc-sleha/assets/3069029/87a8d6bd-3ad0-4a57-9e28-20a72455a485)
![image](https://github.com/SUSE/doc-sleha/assets/3069029/ed031be8-58d9-4b68-9cf5-a7d55fcfa94a)

nicholasyang2022 commented:
It is great to group crmsh operations into 2 categories. However, I think

> full privileges for SSH operations such as `crm  cluster init`, `crm cluster join`, `crm report`.

is not so accuracy. The real privileges needed are read/write access to low-level configurations files, such as `corosync.conf`, and to control system-wide systemd instance to start/stop cluster components such as `corosync` and `pacemaker`.

And SSH operations are needed for crmsh to perform the above actions among different cluster nodes.

On the contrary, cib is owned by hacluster user, and is synchronized across the cluster by pacemaker. So hacluster can operates it and not ssh operations are needed.

Loop in @liangxin1300 and @zzhou1 here. Please review the above description, and see whether it is necessary to include these details in documents.

---

> If you initialize the cluster manually instead of using the `crm clsuter` scripts, to allow SSH agent forwarding you must create the file `/root/.config/crm/crm.conf`

I think creating `crm.conf` manually is error-prone, and is not enough when the public key has not been installed to `authorized_keys`. How about guide to use `crm cluster init ssh --use-ssh-agent` and `crm cluster join ssh --use-ssh-agent` here?

And similarly in [Section 4.8 Bring the cluster online](https://documentation.suse.com/sle-ha/15-SP5/html/SLE-HA-all/cha-ha-ycluster.html#sec-ha-installation-start), `crm cluster start --all` is used. However, this requires SSH operations, which is not available if the cluster is configured using yast2-cluster.

zzhou1 commented:
> It is great to group crmsh operations into 2 categories. However, I think
> 
> > full privileges for SSH operations such as `crm  cluster init`, `crm cluster join`, `crm report`.
> 
> is not so accuracy. The real privileges needed are read/write access to low-level configurations files, such as `corosync.conf`, and to control system-wide systemd instance to start/stop cluster components such as `corosync` and `pacemaker`.
> 
> And SSH operations are needed for crmsh to perform the above actions among different cluster nodes.
> 
> On the contrary, cib is owned by hacluster user, and is synchronized across the cluster by pacemaker. So hacluster can operates it and not ssh operations are needed.
> 
> Loop in @liangxin1300 and @zzhou1 here. Please review the above description, and see whether it is necessary to include these details in documents.
> 

Yeah. Could be something like as below in my language,

"""
These users has full privileges on all cluster nodes by leveraging SSH for crmsh operations such as `crm  cluster init`, `crm cluster join`, `crm report`
"""

> > If you initialize the cluster manually instead of using the `crm clsuter` scripts, to allow SSH agent forwarding you must create the file `/root/.config/crm/crm.conf`
> 
> I think creating `crm.conf` manually is error-prone, and is not enough when the public key has not been installed to `authorized_keys`. How about guide to use `crm cluster init ssh --use-ssh-agent` and `crm cluster join ssh --use-ssh-agent` here?
> 
> And similarly in [Section 4.8 Bring the cluster online](https://documentation.suse.com/sle-ha/15-SP5/html/SLE-HA-all/cha-ha-ycluster.html#sec-ha-installation-start), `crm cluster start --all` is used. However, this requires SSH operations, which is not available if the cluster is configured using yast2-cluster.

Yes, use the "ssh" stage is a better and safer approach.

tahliar commented:
Thanks @zzhou1 and @nicholasyang2022! I've made some changes that hopefully address all your feedback. 
(I'm still working on changing Appendix D, so that's still to come.)

**1. Added some detail about the `crm cluster start` command in [4.8 Bringing the cluster online](https://documentation.suse.com/sle-ha/15-SP5/html/SLE-HA-all/cha-ha-ycluster.html#sec-ha-installation-start)**

![image](https://github.com/SUSE/doc-sleha/assets/3069029/235d21a7-2bf0-4e05-9b29-3cc533202cf7)

I think it's fine to leave in `--all` as long as that clarification is there, because the requirements in chapter 2 already mention that passwordless SSH is required. On that topic:

**2. Added some detail to the SSH requirement in [2.4 Other requirements and recommendations](https://documentation.suse.com/sle-ha/15-SP5/html/SLE-HA-all/cha-ha-requirements.html#sec-ha-requirements-other)**

![image](https://github.com/SUSE/doc-sleha/assets/3069029/eaf7c03c-466d-4bdf-8aea-6e3ecc10e01d)

**3. Updated the new section `5.5.1 Logging in` based on your feedback. It now looks like this:**

![image](https://github.com/SUSE/doc-sleha/assets/3069029/1708b434-99bd-4d79-8d83-f749caf04b3d)
![image](https://github.com/SUSE/doc-sleha/assets/3069029/9ece7732-61bc-4937-a268-820e7279a8cd)

tahliar commented:
I've now also updated [Appendix D](https://documentation.suse.com/sle-ha/15-SP5/html/SLE-HA-all/app-crmreport-nonroot.html). Here's what I did: 

**1. Removed `D.1 Creating a local user account` and `D.2 Configuring a passwordless SSH account`, as they weren't HA-specific and/or are covered elsewhere**

**2. Removed `D.4 Generating a cluster report` as the content is already included elsewhere, and moved the custom SSH port procedure to the `crm report` section in the [Troubleshooting chapter](https://documentation.suse.com/sle-ha/15-SP5/html/SLE-HA-all/app-ha-troubleshooting.html):**

![image](https://github.com/SUSE/doc-sleha/assets/3069029/cf3c95a7-f222-4f9c-98b1-c29c818c1be0)

**3. Consolidated Appendix D down to just the sudo privileges part (PDF because it's a whole section):**

- [app-crmreport-nonroot_en.pdf](https://github.com/SUSE/doc-sleha/files/14927916/app-crmreport-nonroot_en.pdf)

**4. Updated the [regulatory requirements note](https://documentation.suse.com/sle-ha/15-SP5/html/SLE-HA-all/cha-ha-requirements.html#sec-ha-requirements-other) (again) to (re)include a link to the Appendix:**

![image](https://github.com/SUSE/doc-sleha/assets/3069029/0e8e6861-4d3e-451a-a6ba-4a76bf90c97e)

nicholasyang2022 commented:
LGTM :+1:

zzhou1 commented:
> **3. Consolidated Appendix D down to just the sudo privileges part (PDF because it's a whole section):**

Only one comment: 

It's better to remove the space before the asterisk as below. 
`Cmnd_Alias HA_ALLOWED = /bin/su, /usr/sbin/crm report*`


Otherwise, I get the following confusing error message and spend a couple of hours to figure it out

```
hareport@15sp6-1:~> sudo crm report                                                                          
[sudo] password for hareport: 
Sorry, user hareport is not allowed to execute '/usr/sbin/crm report' as root on 15sp6-1.
```

zzhou1 reviewed (changes requested):
Great effort! Only one minor change request.

tahliar commented:
Thanks @nicholasyang2022 and @zzhou1! I've fixed the extra space and will move this on to editorial review now :+1:

dariavladykina commented on xml/ha_config_cli.xml:
```diff
@@ -99,6 +60,127 @@
+      If you set up the cluster with &yast; instead of using the bootstrap scripts, the SSH keys
+      are not automatically generated. To configure SSH keys with SSH agent forwarding enabled,
+      you can use the <literal>ssh</literal> stage of the bootstrap scripts on its own. Run these
+      commands after configuring the cluster in &yast;, but before bringing the cluster online:
```
```suggestion
      commands after configuring the cluster in &yast;, but before bringing the cluster online.
```

dariavladykina reviewed (commented):
LGTM, just one minor nit. Thanks!
